### DISCUSSION TITLE
System Navigation Manager implemented in UWP, Android, IOS, but NOT in WINUI3

### DISCUSSION THREAD

**Discussion by cesarchefinho** [unoplatform/uno#8526] [Q&A]
System Navigation Manager is not implemented in Winui3, this prevent migrate my projects frow Uno uwp version to uno winui version and handle hardware back button in winui3

**Comment by jeromelaban** [comment]
Thanks for the report. Can you specify which platform you are talking about for WinUI 3? Is it WinAppSDK on Windows?

**Comment by cesarchefinho** [comment]
Yes. SystemNavigationManager is not implemented in winappsdk (windows) but is implemented in Android and iOS and webasm Winui3 plataforms

**Reply by jeromelaban** [reply]
Thanks. In this case, WinAppSDK is not something Uno can change, as it's managed by Microsoft. There's no support for hardware back button in Windows (for 11 and up) and it was removed/deprecated when migrating to WinAppSDK.

If you need a "back" management, you'll need to integrate this with an in-app back button.

**Reply by cesarchefinho** [reply]
But uno can implement this class in winui and intercept wm-back keyboard and mouse back key and with this my code that use SystemNavigationManager will work across all Uno platforms. Android don't support SystemNavigationManager and Uno inplement this class, wy not implement in Winui3?

**Reply by jeromelaban** [reply]
If you're running the Windows head, there's no Uno available in this context, so we cannot add anything to Microsoft's code.

**Reply by FrozDark** [reply]
> But uno can implement this class in winui and intercept wm-back keyboard and mouse back key and with this my code that use SystemNavigationManager will work across all Uno platforms. Android don't support SystemNavigationManager and Uno inplement this class, wy not implement in Winui3?

Uno is a separate framework apart from WinUI (WinAppSDK). The later one is handled by the Microsoft and there is nothing to do from Uno side to modify the Windows head. Uno is just using the same code that had been done by Microsoft (WinUI) and provides underlying platforms and frameworks (Android, iOS, Mac, WPF, WASM) to look and feel the same as Windows head

**Comment by MartinZikmund** [comment]
@cesarchefinho You can still use the class in Uno Platform apps by wrapping it in `#if HAS_UNO` - this way the code will be compiled only on Uno targets
